libs/common/src/token/trade.service.ts

```diff
@@ -0,0 +1,66 @@
+import { HttpService } from '@nestjs/axios'
+import { Injectable, Inject } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
+import { lastValueFrom } from 'rxjs'
+import { NATIVE_FUSE_TOKEN } from '@app/smart-wallets-service/common/constants/fuseTokenInfo'
+import { CACHE_MANAGER } from '@nestjs/cache-manager'
+import { Cache } from 'cache-manager'
+
+@Injectable()
+export default class TradeService {
+  constructor (
+    private readonly httpService: HttpService,
+    private readonly configService: ConfigService,
+    @Inject(CACHE_MANAGER) private cacheManager: Cache
+  ) { }
+
+  private readonly API_URL = 'https://pro-api.coingecko.com/api/v3'
+
+  async getTokenPriceByAddress (tokenAddress: string): Promise<number> {
+    const cacheKey = `token_price_${tokenAddress.toLowerCase()}`
+    const cachedPrice = await this.cacheManager.get<number>(cacheKey)
+
+    if (cachedPrice !== undefined) {
+      return cachedPrice
+    }
+
+    const apiKey = this.configService.get<string>('coinGeckoApiKey')
```

Comment:
This can be impoved by utilizing the `getOrThrow` method. E.g: `this.configService.getOrThrow('coinGeckoApiKey')`